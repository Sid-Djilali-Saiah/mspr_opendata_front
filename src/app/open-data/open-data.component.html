<div class="app-container" *ngIf="data$ | async as recipesList">
  <mat-card class='mat-elevation-z2'>
    <h2 class='mat-h1'>Import Data</h2>
    <angular-file-uploader
      [config]="afuConfig"
      [resetUpload]='true'
    >
    </angular-file-uploader>
  </mat-card>
  <mat-card class='mat-elevation-z2' style='margin-top: 30px'>
    <h2 class='mat-h1'>Uploaded Data</h2>
    <mat-divider></mat-divider>
    <mat-selection-list multiple [(ngModel)]="selectedOptions">
      <mat-list-option *ngFor="let recipe of recipesList" [value]="recipe.id">
        {{ recipe.name }}
      </mat-list-option>
    </mat-selection-list>
    <mat-divider></mat-divider>
    <div class="button-container">
      <button mat-stroked-button color="primary" [disabled]="selectedOptions.length <= 0" (click)="download()">
        Download
      </button>
    </div>
  </mat-card>
</div>
