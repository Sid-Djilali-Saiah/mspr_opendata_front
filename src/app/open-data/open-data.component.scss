.app-container {
  margin: 40px auto;
  height: 100%;
  max-width: 1180px;
}

.button-container {
  display: flex;
  justify-content: flex-end;
  margin-top: 30px;
}
