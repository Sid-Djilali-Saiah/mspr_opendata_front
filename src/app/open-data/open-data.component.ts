import { Component, OnInit } from '@angular/core';
import { OpenDataService } from '../core/services/open-data.service';
import { Observable } from 'rxjs';
import {UntilDestroy, untilDestroyed} from "@ngneat/until-destroy";
import { AngularFileUploaderConfig } from 'angular-file-uploader';

@UntilDestroy()
@Component({
  selector: 'app-recipe-page',
  templateUrl: './open-data.component.html',
  styleUrls: ['./open-data.component.scss'],
})
export class OpenDataComponent implements OnInit {

  afuConfig: AngularFileUploaderConfig = {
    multiple: false,
    formatsAllowed: ".csv",
    maxSize: 5,
    uploadAPI:  {
      url:"https://api-opendata.nonstopintegration.ml/api/opendata/import",
      method:"POST",
      responseType: 'application/json'
    }
  };
  data$: Observable<any>;
  selectedOptions = [];

  constructor(private openDataService: OpenDataService) {}

  ngOnInit(): void {
    this.data$ = this.openDataService.list();
  }

  download(): void {
    if (this.selectedOptions.length > 0) {
      this.openDataService.download(this.selectedOptions).pipe(
        untilDestroyed(this),
      ).subscribe(data => {
        const link: HTMLAnchorElement = window.document.createElement('a');
        link.href = URL.createObjectURL(data);
        link.target = '_blank';
        link.download = 'data.pdf';
        link.click();
      });
    }
  }
}
