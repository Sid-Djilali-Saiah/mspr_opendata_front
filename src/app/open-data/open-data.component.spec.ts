import { ComponentFixture, TestBed } from '@angular/core/testing';

import { OpenDataComponent } from './open-data.component';
import { RouterTestingModule } from "@angular/router/testing";
import { MaterialModule } from "../shared/material.module";
import { FormsModule } from "@angular/forms";
import {OpenDataService} from "../core/services/open-data.service";
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { AngularFileUploaderModule } from 'angular-file-uploader';

describe('OpenDataComponent', () => {
  let component: OpenDataComponent;
  let fixture: ComponentFixture<OpenDataComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [OpenDataService],
      imports: [
        RouterTestingModule.withRoutes([]),
        MaterialModule,
        FormsModule,
        AngularFileUploaderModule,
        HttpClientTestingModule,
        BrowserAnimationsModule
      ],
      declarations: [OpenDataComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(OpenDataComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should equal snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
