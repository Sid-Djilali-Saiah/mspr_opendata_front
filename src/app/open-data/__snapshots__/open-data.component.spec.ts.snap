// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenDataComponent should equal snapshot 1`] = `
<app-recipe-page
  afuConfig={[Function Object]}
  data$={[Function Observable]}
  openDataService={[Function OpenDataService]}
  selectedOptions={[Function Array]}
>
  
</app-recipe-page>
`;
